Userland/Libraries/LibWeb/Bindings/WindowObject.cpp

```diff
@@ -437,4 +449,167 @@ JS_DEFINE_NATIVE_FUNCTION(WindowObject::inner_height_getter)
     return JS::Value(impl->inner_height());
 }
 
+JS_DEFINE_NATIVE_GETTER(WindowObject::scroll_x_getter)
+{
+    auto* impl = impl_from(vm, global_object);
+    if (!impl)
+        return {};
+    if (!impl->page())
+        return JS::Value(0.0);
+    return JS::Value(static_cast<double>(impl->page()->top_level_browsing_context().viewport_scroll_offset().x()));
+}
+
+JS_DEFINE_NATIVE_GETTER(WindowObject::scroll_y_getter)
+{
+    auto* impl = impl_from(vm, global_object);
+    if (!impl)
+        return {};
+    if (!impl->page())
+        return JS::Value(0.0);
+    return JS::Value(static_cast<double>(impl->page()->top_level_browsing_context().viewport_scroll_offset().y()));
```

Comment:
There's a `Value(i32)` ctor, which should fit here, no? `viewport_scroll_offset()` returns an `IntPoint`.